server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/VariableWidthHistogramAggregator.java

```diff
@@ -179,42 +179,40 @@ public void collect(int doc, long bucket) throws IOException {
      */
     private void collectValue(LeafBucketCollector sub, int doc, double val) throws IOException{
         if (numCachedDocs < cacheLimit) {
-            System.out.println("Cache limit = " + cacheLimit);
-            // Make a new bucket for the document
+            // Cache the doc in a new bucket, rather than trying to add it into an existing bucket
             cachedValues = bigArrays.grow(cachedValues, numCachedDocs + 1);
-            cachedValues.set((long)numCachedDocs, val);
+            cachedValues.set((long) numCachedDocs, val);
             collectBucket(sub, doc, numCachedDocs);
             numCachedDocs += 1;
 
-            if (numCachedDocs == cacheLimit) {
-                // Cache is full -> Cluster the documents to reduce the number of buckets
-                // This will not occur more than once since cacheLimit is never reset
-                bucketCachedDocs();
-
-                // Calculate the average distance between buckets
-                // Subsequent documents will be compared with this value to determine if they should be collected into
-                // an existing bucket or into a new bucket
-                // Assumption: buckets are sorted by centroid
-                int sum = 0;
-                for (int i = 0; i < numClusters - 1; i++) {
-                    sum += clusterCentroids.get(i + 1) - clusterCentroids.get(i);
-                }
-                avgBucketDistance = (sum / (numClusters - 1));
-            }
-        } else {
-            int bucketOrd = getNearestBucket(val);
-            double distance = Math.abs(clusterCentroids.get(bucketOrd)- val);
-            if(bucketOrd == -1 || distance > (2 * avgBucketDistance) && numClusters < shardSize) {
-                // Make a new bucket since the document is distant from all existing buckets
-                // TODO: Create a new bucket even if (# buckets) > shardSize. This will require merging nearby buckets
-                //       after sorting by centroid in buildAggregation(), to ensure at most shardSize buckets are produced
-                addToCluster(numClusters, val);
-                collectBucket(sub, doc, numClusters);
-            } else {
-                addToCluster(bucketOrd, val);
-                collectExistingBucket(sub, doc, bucketOrd);
+            if(numCachedDocs != cacheLimit) return;
```

Comment:
We *tend* to prefer:
```
if (numCacheDocs != cacheLimit) {
  return;
}
```

It so much wider but it makes the `return` just stand out a bit better. Probably just my legacy as a java developer for so many years kicking in.